gems/aws-sdk-core/lib/seahorse/client/plugins/content_length.rb

```diff
@@ -9,14 +9,25 @@ class ContentLength < Plugin
         class Handler < Client::Handler
 
           def call(context)
-            # If it's an IO object and not a File / String / String IO
-            if context.http_request.body.respond_to?(:size)
-              length = context.http_request.body.size
+            body = context.http_request.body
+            if (length = body_size(body))
               context.http_request.headers['Content-Length'] = length
             end
             @handler.call(context)
           end
 
+          def body_set?(body)
```

Comment:
This isn't used.

Also talked offline, let's implement body_size on http_request